nylas/models/messages.py

```diff
@@ -0,0 +1,179 @@
+
+    Attributes:
+        name: Participant's name
+        email: Participant's email
+    """
+
+    email: str
+    name: Optional[str] = None
+
+
+@dataclass_json
+@dataclass
+class Attachment:
+    """
+    An attachment on a message.
+
+    Attributes:
+        id: Globally unique object identifier.
+        size: Size of the attachment in bytes.
+        filename: Name of the attachment.
+        content_type: MIME type of the attachment.
+    """
+
+    id: str
+    size: int
+    filename: Optional[str] = None
+    content_type: Optional[str] = None
+
+
+@dataclass_json
+@dataclass
+class Message:
+    """
+    A Message object.
+
+    Attributes:
+        id: Globally unique object identifier.
+        grant_id: The grant that this message belongs to.
+        thread_id: The thread that this message belongs to.
+        subject: The subject of the message.
+        from_: The sender of the message.
+        to: The recipients of the message.
+        cc: The CC recipients of the message.
+        bcc: The BCC recipients of the message.
+        reply_to: The reply-to recipients of the message.
+        date: The date the message was received.
+        unread: Whether the message is unread.
+        starred: Whether the message is starred.
+        snippet: A snippet of the message body.
+        body: The body of the message.
+        attachments: The attachments on the message.
+        folders: The folders that the message is in.
+        headers: The headers of the message.
+    """
+
+    id: str
+    grant_id: str
+    thread_id: str
+    subject: str
+
+    from_: List[Participant] = field(metadata=config(field_name="from"))
+    to: List[Participant]
+
+    date: datetime
+
+    unread: bool
+    starred: bool
+
+    snippet: str
+    body: str
+
+    bcc: Optional[List[Participant]] = None
+    cc: Optional[List[Participant]] = None
+    reply_to: Optional[List[Participant]] = None
+    attachments: Optional[List[Attachment]] = None
+    folders: Optional[List[str]] = None
+    headers: Optional[List[str]] = None
+
+
+# Need to use Functional typed dicts because "from" and "in" are Python
+# keywords, and can't be declared using the declarative syntax
+ListMessagesQueryParams = TypedDict(
+    "ListMessagesQueryParams",
+    {
+        **get_type_hints(ListQueryParams),  # Inherit fields from ListQueryParams
+        "subject": NotRequired[str],
+        "any_email": NotRequired[str],
+        "from": NotRequired[str],
+        "to": NotRequired[str],
+        "cc": NotRequired[str],
+        "bcc": NotRequired[str],
+        "in": NotRequired[str],
+        "unread": NotRequired[bool],
+        "starred": NotRequired[bool],
+        "thread_id": NotRequired[str],
+        "received_before": NotRequired[int],
+        "received_after": NotRequired[int],
+        "has_attachment": NotRequired[bool],
+        "fields": NotRequired[Fields],
+        "search_query_native": NotRequired[str],
+    },
+)
+"""
```

Comment:
FYI, I have no idea if this docstring will end up in the autogenerated docs. Couldn't figure out any way to include a docstring on a functionally declared TypedDict.